### PR TITLE
clickhouse: 23.11.1.2711 -> 24.3.2.23

### PR DESCRIPTION
**Pull request by mbalatsko**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build clickhouse

File: pkgs/servers/clickhouse/default.nix [hunk 1]
```diff
@@ -96,24 +98,33 @@ in mkDerivation rec {
...
-    preBuild = "cd generator";
-    hash = "sha256-dhUgpwSjE9NZ2mCkhGiydI51LIOClA5wwk1O3mnnbM8=";
-  } else null;
-  rustDeps = if rustSupport then rustPlatform.fetchCargoTarball {
-    inherit src;
-    name = "rust-deps";
-    preBuild = "cd rust";
-    hash = "sha256-fWDAGm19b7uZv8aBdBoieY5c6POd8IxFXbGdtONpZbw=";
-  } else null;
+  corrosionSrc =
```
Comment (SuperSandro2000) [inline pkgs/servers/clickhouse/default.nix:101]: Please keep the old formatting

Comment (mbalatsko) [reply pkgs/servers/clickhouse/default.nix]: fixed

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build clickhouse

File: pkgs/servers/clickhouse/default.nix [hunk 2]
```diff
@@ -206,7 +212,7 @@ in mkDerivation rec {
     homepage = "https://clickhouse.com";
     description = "Column-oriented database management system";
     license = licenses.asl20;
-    maintainers = with maintainers; [ orivej ];
+    maintainers = with maintainers; [ orivej mbalatsko ];
```
Comment (ixmatus) [inline pkgs/servers/clickhouse/default.nix:215]: @mbalatsko IIRC you need to be in a list of maintainers elsewhere in nixpkgs to add yourself here. If you are I'm fine with this change but if not, you may get an evaluation error.

Comment (mbalatsko) [reply pkgs/servers/clickhouse/default.nix:215]: I am a maintainer already ;)

**Review by ixmatus — APPROVED** [review]
LGTM

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build clickhouse

**Comment by mbalatsko** [comment]
Result of `nixpkgs-review pr 301642` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>clickhouse</li>
  </ul>
</details>

**Comment by mbalatsko** [comment]
Result of `nixpkgs-review pr 301642` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>clickhouse</li>
  </ul>
</details>

**Comment by risicle** [comment]
#301689 was this just a coincidence?

**Comment by mbalatsko** [comment]
@risicle oh, interesting! I had no intent to duplicate, haven't seen your PR

**Pull request merged**